docling_core/types/doc/page.py

```diff
@@ -0,0 +1,1292 @@
+
+        Returns:
+            Updated ImageDraw object
+        """
+        fill = self._get_rgba(name=line_color, alpha=line_alpha)
+
+        # Draw each rectangle by connecting its four points
+        for line in self.lines:
+
+            line.to_top_left_origin(page_height=page_height)
+            for segment in line.iterate_segments():
+                draw.line(
+                    (segment[0][0], segment[0][1], segment[1][0], segment[1][1]),
+                    fill=fill,
+                    width=max(1, round(line.width)),
+                )
+
+        return draw
+
+
+class PdfMetaData(BaseModel):
+    """Model representing PDF metadata extracted from XML."""
+
+    xml: str = ""
+
+    data: Dict[str, str] = {}
+
+    def initialise(self):
+        """Initialize metadata by parsing the XML content."""
+        # Define the regex pattern
+        pattern = r"\<([a-zA-Z]+)\:([a-zA-Z]+)\>(.+?)\<\/([a-zA-Z]+)\:([a-zA-Z]+)\>"
+
+        # Find all matches
+        matches = re.findall(pattern, self.xml)
+
+        # Process matches
+        for _ in matches:
+            namespace_open, tag_open, content, namespace_close, tag_close = _
+            if namespace_open == namespace_close and tag_open == tag_close:
+                print(
+                    f"Namespace: {namespace_open}, Tag: {tag_open}, Content: {content}"
+                )
+                self.data[tag_open] = content
+
+
+class PdfTableOfContents(BaseModel):
+    """Model representing a PDF table of contents entry with hierarchical structure."""
+
+    text: str
+    orig: str = ""
+
+    marker: str = ""
+
+    children: List["PdfTableOfContents"] = []
+
+    def export_to_dict(self, mode: str = "json") -> Dict:
+        """Export the table of contents to a dictionary.
+
+        Args:
+            mode: Serialization mode
+
+        Returns:
+            Dictionary representation of the table of contents
+        """
+        return self.model_dump(mode=mode, by_alias=True, exclude_none=True)
+
+    def save_as_json(self, filename: Path, indent: int = 2):
+        """Save the table of contents as a JSON file.
+
+        Args:
+            filename: Path to save the JSON file
+            indent: Indentation level for JSON formatting
+        """
+        out = self.export_to_dict()
+        with open(filename, "w", encoding="utf-8") as fw:
+            json.dump(out, fw, indent=indent)
+
+    @classmethod
+    def load_from_json(cls, filename: Path) -> "PdfTableOfContents":
+        """Load table of contents from a JSON file.
+
+        Args:
+            filename: Path to the JSON file
+
+        Returns:
+            Instantiated PdfTableOfContents object
+        """
+        with open(filename, "r", encoding="utf-8") as f:
+            return cls.model_validate_json(f.read())
+
+
+class ParsedPdfDocument(BaseModel):
+    """Model representing a completely parsed PDF document with all components."""
+
+    pages: Dict[int, SegmentedPdfPage] = {}
```

Comment:
Let's make sure the integer is always >=1

pages: Dict[conint(ge=1), SegmentedPdfPage] = {}